src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -198,6 +198,26 @@ int sd_dhcp6_client_set_duid(sd_dhcp6_client *client, uint16_t duid_type,
         return 0;
 }
 
+int sd_dhcp6_client_get_duid(sd_dhcp6_client *client, uint16_t *duid_type,
+                             const uint8_t **duid, size_t *duid_len) {
+        assert_return(client, -EINVAL);
+        assert_return(duid_type, -EINVAL);
+        assert_return(duid_len, -EINVAL);
+        assert_return(duid, -EINVAL);
```

Comment:
This follows the same paradigm as its IPv4 counterpart: sd_dhcp_client_get_client_id. I believe it is safe to assume that the intent of calling get API is to query information from it, and thus pass valid out-parameters to meet that need. We can certainly do the extra error-checking here. Or just have the caller do it (and save the cost of a function call). 

Is there any advantage in allowing caller to pass in NULL and have the ability to call the API and not get anything out of it?